In [1]:
import pandas as pd
import numpy as np
import json

Importing the data as a dataframe.

In [34]:
dataset = pd.read_csv("data/Data_Lv3_UMDOTS_Escooters.csv")

Lets take a look at a few data point

In [84]:
#dataset.head()
dataset.tail()

Unnamed: 0  Unnamed: 0.1                   CREATED  \
40320       40320         40320 2020-10-24 22:21:40+00:00   
40321       40321         40321 2020-10-24 21:28:43+00:00   
40322       40322         40322 2020-10-25 01:22:00+00:00   
40323       40323         40323 2020-10-25 00:55:24+00:00   
40324       40324         40324 2020-10-24 06:03:05+00:00   

                          START                       END  DISTANCE  RIDE_ID  \
40320 2020-10-24 22:21:40+00:00 2020-10-24 22:26:38+00:00     0.794  3304114   
40321 2020-10-24 21:28:43+00:00 2020-10-24 21:32:54+00:00     0.404  3303630   
40322 2020-10-25 01:22:00+00:00 2020-10-25 01:26:16+00:00     0.629  3305589   
40323 2020-10-25 00:55:24+00:00 2020-10-25 01:07:38+00:00     0.567  3305336   
40324 2020-10-24 06:03:05+00:00 2020-10-24 06:38:05+00:00     5.220  3299791   

       CUSTOMER_ID  MINUTES VEHICLE_TYPE  \
40320       322401        5      Scooter   
40321       298095        5      Scooter   
40322       604319        5      Scooter   
40323       610943       13      Scooter   
40324       324757       35      Scooter   

                                                    PATH  \
40320  {\n  "coordinates": [\n    [\n      -76.943936...   
40321  {\n  "coordinates": [\n    [\n      -76.935917...   
40322  {\n  "coordinates": [\n    [\n      -76.940231...   
40323  {\n  "coordinates": [\n    [\n      -76.944152...   
40324  {\n  "coordinates": [\n    [\n      -76.934166...   

                                              TIMESTAMPS  START LONG  \
40320  [2020-10-24T22:21:46Z, 2020-10-24T22:22:02Z, 2...  -76.943937   
40321  [2020-10-24T21:28:49Z, 2020-10-24T21:29:05Z, 2...  -76.935917   
40322  [2020-10-25T01:22:07Z, 2020-10-25T01:22:18Z, 2...  -76.940231   
40323  [2020-10-25T00:55:30Z, 2020-10-25T00:55:46Z, 2...  -76.944153   
40324  [2020-10-24T06:03:12Z, 2020-10-24T06:05:12Z, 2...  -76.934167   

       START LAT   END LONG    END LAT  Year  
40320  38.988110 -76.934104  38.992061  2020  
40321  38.986389 -76.933818  38.992196  2020  
40322  38.996377 -76.945600  38.992842  2020  
40323  38.991956 -76.940216  38.987199  2020  
40324  38.990645 -76.936088  38.982452  2020

In [19]:
dataset.shape

(40325, 16)

In [44]:
dataset.dtypes

Unnamed: 0                    int64
Unnamed: 0.1                  int64
CREATED         datetime64[ns, UTC]
START           datetime64[ns, UTC]
END             datetime64[ns, UTC]
DISTANCE                    float64
RIDE_ID                       int64
CUSTOMER_ID                   int64
MINUTES                       int64
VEHICLE_TYPE                 object
PATH                         object
TIMESTAMPS                   object
START LONG                  float64
START LAT                   float64
END LONG                    float64
END LAT                     float64
dtype: object

In [51]:
# All the data types do not match lets fix them

# Convert as Timestamp data
dataset["CREATED"] = pd.to_datetime(dataset["CREATED"])
# Convert as Timestamp data
dataset["START"] = pd.to_datetime(dataset["START"])
# Convert as Timestamp data
dataset["END"] = pd.to_datetime(dataset["END"])
# Convert timestamp as list
dataset["TIMESTAMPS"] = dataset["TIMESTAMPS"].apply(eval)

The dataset has 40325 rows and 16 columns

In [96]:
dataset["Year"][12466]

2019

In [95]:
# Get the year of each data entry
for index, item in dataset.iterrows():
    dataset["Year"][index] = dataset["CREATED"][index].year

<ipython-input-95-5d6f9b1b9c5e>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset["Year"][index] = dataset["CREATED"][index].year


In [97]:
ride_count_year = dataset.groupby("Year").size().reset_index(name='ride_count')
ride_count_year

Year  ride_count
0  2019       18382
1  2020       21943

The distance column is missing values

In [12]:
dataset["DISTANCE"].isna().sum()
#dataset["DISTANCE"].dropna()

2273

Total of 2273 missing datapoint for distance. No sure how to deal with this yet.

Next we separate all ride by years: October 2020 and October 2019

In [57]:
dataset.groupby("CREATED")